src/vanilla.jl

```diff
@@ -132,30 +132,33 @@ function solve(solver::ValueIterationSolver, mdp::Union{MDP,POMDP}, policy=creat
         residual = 0.0
         tic()
         # state loop
-        for (istate, s) in enumerate(iterator(sspace))
-            old_util = util[istate] # for residual 
+        for s in iterator(sspace)
+            sidx = state_index(mdp, s)
+            old_util = util[sidx] # for residual 
             max_util = -Inf
             # action loop
             # util(s) = max_a( R(s,a) + discount_factor * sum(T(s'|s,a)util(s') )
-            for (iaction, a) in enumerate(iterator(aspace))
+            sub_aspace = POMDPs.actions(mdp, s, aspace)
```

Comment:
Where is sub_aspace used? I couldn't find it below. Should aspace be replaced with sub_aspace in 142